pkg/kube/config_getter.go

```diff
@@ -35,7 +35,14 @@ func NewImpersonatingConfigGetter(cfg *rest.Config, insecure bool) *Impersonatin
 func (r *ImpersonatingConfigGetter) Config(ctx context.Context) *rest.Config {
 	shallowCopy := *r.cfg
 
-	if p := auth.Principal(ctx); p != nil {
+	var hasToken bool
+
+	if t := auth.BearerToken(ctx); len(t) != 0 {
+		shallowCopy.BearerToken = t
+		hasToken = true
+	}
```

Comment:
> The way this is written, I think the insecure flag only works if you didn't provide a token, which doesn't seem right.

The insecure flag will still work, it only bypasses the Principal check, although I am about to fix that. I believe @bigkevmcd is also going to be removing the insecure check in a later PR.

> Finally, I'm a bit confused. I thought the code we use for impersonation was in clustersmngr.go

The client in clustermgr.go is, I believe, mainly used for communication with leaf clusters from the wego server.

Reply:
> I wish there was an on-off switch, rather than something everybody can always do.

For sure, we could definitely enable / disable token passthrough via a flag etc here?